gctscript/modules/gct/exchange.go

```diff
@@ -0,0 +1,381 @@
+package gct
+
+import (
+	"strings"
+
+	"github.com/d5/tengo/objects"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/withdraw"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/wrappers"
+)
+
+var exchangeModule = map[string]objects.Object{
+	"orderbook":      &objects.UserFunction{Name: "orderbook", Value: ExchangeOrderbook},
+	"ticker":         &objects.UserFunction{Name: "ticker", Value: ExchangeTicker},
+	"exchanges":      &objects.UserFunction{Name: "exchanges", Value: ExchangeExchanges},
+	"pairs":          &objects.UserFunction{Name: "pairs", Value: ExchangePairs},
+	"accountinfo":    &objects.UserFunction{Name: "accountinfo", Value: ExchangeAccountInfo},
+	"depositaddress": &objects.UserFunction{Name: "depositaddress", Value: ExchangeDepositAddress},
+	"orderquery":     &objects.UserFunction{Name: "orderquery", Value: ExchangeOrderQuery},
+	"ordercancel":    &objects.UserFunction{Name: "ordercancel", Value: ExchangeOrderCancel},
+	"ordersubmit":    &objects.UserFunction{Name: "ordersubmit", Value: ExchangeOrderSubmit},
+	"withdrawcrypto": &objects.UserFunction{Name: "withdrawcrypto", Value: ExchangeWithdrawCrypto},
+	"withdrawfiat":   &objects.UserFunction{Name: "withdrawfiat", Value: ExchangeWithdrawFiat},
+}
+
+// ExchangeOrderbook returns orderbook for requested exchange & currencypair
+func ExchangeOrderbook(args ...objects.Object) (objects.Object, error) {
+	if len(args) != 4 {
+		return nil, objects.ErrWrongNumArguments
+	}
+
+	exchangeName, _ := objects.ToString(args[0])
+	currencyPair, _ := objects.ToString(args[1])
+	delimiter, _ := objects.ToString(args[2])
+	assetTypeParam, _ := objects.ToString(args[3])
+
+	pairs := currency.NewPairDelimiter(currencyPair, delimiter)
+	assetType := asset.Item(assetTypeParam)
+
+	ob, err := wrappers.GetWrapper().Orderbook(exchangeName, pairs, assetType)
```

Comment:
engine helpers.go needs to handle a nil pointer return from GetExchangeByName in GetSpeicifcOrderbook ticker etc, if exchange isn't enabled